healthcheck.php

```diff
@@ -0,0 +1,74 @@
+<?php
+namespace Automattic\VIP;
+
+/**
+ * Utility Healthcheck class that runs further than cache-healthcheck,
+ * and allows to verify the app code is working as expected.
+ */
+class Healthcheck {
+	/**
+	 * By default, an app is considered healthy unless filter value returns a non-empty array
+	 *
+	 * @var array
+	 */
+	protected array $errors = [];
+
+	public function __construct() {
+		$this->check();
+	}
+
+	/**
+	 * Apply the filters to and set the `$errors` property accordingly.
+	 *
+	 * @return void
+	 */
+	public function check() {
+		/**
+		 * Filters the errors that would indicate the app is unhealthy.
+		 *
+		 * @param string[]     $errors either empty array (no errors) or array of strings: [ 'an error has occurred', 'another error' ]
+		 */
+		$this->errors = apply_filters( 'vip_site_healthcheck_errors_array', $this->errors );
```

Comment:
What sorts of things do we expect customer's to add in checks for? Asking because maybe we should ship this endpoint initially without a filter like this. If/once the use case comes up where there is some sort of dependency to people want to check, we'd maybe end up going a different route with how we want to handle it.

Reply:
> What sorts of things do we expect customer's to add in checks for?

Not sure just yet, I think that hugely depends on doc coverage and our advice in implementing it, the goal here was to implement something more advanced from the get-go. There are plenty of ways someone can break their app with PHP alone which would otherwise be considered "healthy" by our simple "if it loaded the codebase it's ok" check. So I thought providing some flexibility by allowing to check something critical for the app but not resulting in a fatal would be nice.

Reply:
@simonwheatley @mjangda thoughts on this?

Reply:
My thought process is that it's always easier to add a filter later, but much harder to remove one. And particularly in this case, I'm not able to think of any really good use cases - but a number of bad ones 🙃 .

A "readiness" check to me just means that the application built (no fatals), and required services are online/connected (DB/memcache). Since this endpoint returns after all the `init` stuff takes place, I think we cover both pretty well.

If a customer wants a particular "ensure this is present before a deploy" check, then at any point during the `init` process they could throw a PHP fatal.